doc/development/create-a-model-pt.md

```diff
@@ -0,0 +1,161 @@
+        pass
+
+    def get_ntypes(self) -> int:
+        pass
+
+    def get_dim_out(self) -> int:
+        pass
+
+    def get_dim_emb(self) -> int:
+        pass
+
+    def mixed_types(self) -> bool:
+        pass
+
+    def forward(
+        self,
+        coord_ext: torch.Tensor,
+        atype_ext: torch.Tensor,
+        nlist: torch.Tensor,
+        mapping: Optional[torch.Tensor] = None,
+    ):
+        pass
+
+    def serialize(self) -> dict:
+        pass
+
+    def deserialize(cls, data: dict) -> "SomeDescript":
+        pass
+
+    def update_sel(cls, global_jdata: dict, local_jdata: dict):
+        pass
+```
+
+The serialize and deserialize methods are important for cross-backend model conversion.
+
+### New fitting nets
+
+In many instances, there is no requirement to create a new fitting net. For fitting user-defined scalar properties, the {py:class}`deepmd.pt.model.task.ener.InvarFitting` class can be utilized. However, if there is a need for a new fitting net, one should inherit from both the {py:class}`deepmd.pt.model.task.base_fitting.BaseFitting` class and the {py:class}`torch.nn.Module` class. Alternatively, for a more straightforward approach, inheritance from the {py:class}`deepmd.pt.model.task.fitting.GeneralFitting` class is also an option.
+
+
+```py
+from deepmd.pt.model.task.fitting import (
+    GeneralFitting,
+)
+from deepmd.dpmodel import (
+    FittingOutputDef,
+    fitting_check_output,
+)
+
+
+@GeneralFitting.register("some_fitting")
+@fitting_check_output
+class SomeFittingNet(GeneralFitting):
+    def __init__(self, arg1: bool, arg2: float) -> None:
+        pass
+
+    def forward(
+        self,
+        descriptor: torch.Tensor,
+        atype: torch.Tensor,
+        gr: Optional[torch.Tensor] = None,
+        g2: Optional[torch.Tensor] = None,
+        h2: Optional[torch.Tensor] = None,
+        fparam: Optional[torch.Tensor] = None,
+        aparam: Optional[torch.Tensor] = None,
+    ):
+        pass
+
+    def output_def(self) -> FittingOutputDef:
+        pass
+```
+### New models
+The PyTorch backend's model architecture is meticulously structured with multiple layers of abstraction, ensuring a high degree of flexibility. Typically, the process commences with an atomic model responsible for atom-wise property calculations. This atomic model inherits from both the {py:class}`deepmd.pt.model.atomic_model.base_atomic_model.BaseAtomicModel` class and the {py:class}`torch.nn.Module` class.
+
+Subsequently, the `AtomicModel` is encapsulated using the `make_model(AtomicModel)` function, which leverages the `deepmd.pt.model.model.make_model.make_model` function.  The purpose of the `make_model` wrapper is to facilitate the translation between atomic property predictions and the extended property predictions and differentiation , e.g. the reduction of atomic energy contribution and the autodiff for calculating the forces and virial. The developers usually need to implement an `AtomicModel` not a `Model`.
```

Comment:
shall we add the inheritance of forward functions in different level? e.g. forward --> forward_common --> forward_common_lower --> forward_common_atomic --> forward_atomic

Reply:
As discussed, in most cases the developer only need to handle AtomicModel, thus only cares about `forward_atomic`.